src/instana/instrumentation/grpcio.py

```diff
@@ -33,248 +38,308 @@ def collect_tags(span, instance, argv, kwargs):
                 method = argv[2][2][1]._method.decode()
                 target = argv[2][2][1]._channel.target().decode()
 
-            span.set_tag('rpc.call', method)
+            span.set_attribute("rpc.call", method)
 
-            if ':///' in target:
-                _, target, *_ =  target.split(':///')
-            parts = target.split(':')
+            if ":///" in target:
+                _, target, *_ = target.split(":///")
+            parts = target.split(":")
             if len(parts) == 2:
-                span.set_tag('rpc.host', parts[0])
-                span.set_tag('rpc.port', parts[1])
-        except:
-            logger.debug("grpc.collect_tags non-fatal error", exc_info=True)
+                span.set_attribute("rpc.host", parts[0])
+                span.set_attribute("rpc.port", parts[1])
+        except Exception:
+            logger.debug("grpc.collect_attributes non-fatal error", exc_info=True)
         return span
 
-
-    @wrapt.patch_function_wrapper('grpc._channel', '_UnaryUnaryMultiCallable.with_call')
+    @wrapt.patch_function_wrapper("grpc._channel", "_UnaryUnaryMultiCallable.with_call")
     def unary_unary_with_call_with_instana(wrapped, instance, argv, kwargs):
-        parent_span = tracer.active_span
+        parent_span = get_current_span()
 
         # If we're not tracing, just return
-        if parent_span is None:
+        if not parent_span.is_recording():
             return wrapped(*argv, **kwargs)
 
-        with tracer.start_active_span("rpc-client", child_of=parent_span) as scope:
+        parent_context = parent_span.get_span_context() if parent_span else None
+
+        with tracer.start_as_current_span(
+            "rpc-client", span_context=parent_context
+        ) as span:
             try:
                 if "metadata" not in kwargs:
                     kwargs["metadata"] = []
 
-                kwargs["metadata"] = tracer.inject(scope.span.context, opentracing.Format.BINARY, kwargs['metadata'],
-                                                   disable_w3c_trace_context=True)
-                collect_tags(scope.span, instance, argv, kwargs)
-                scope.span.set_tag('rpc.call_type', 'unary')
+                kwargs["metadata"] = tracer.inject(
+                    span.context,
+                    Format.BINARY,
+                    kwargs["metadata"],
+                    disable_w3c_trace_context=True,
+                )
+                collect_attributes(span, instance, argv, kwargs)
+                span.set_attribute("rpc.call_type", "unary")
 
                 rv = wrapped(*argv, **kwargs)
             except Exception as e:
-                scope.span.log_exception(e)
+                span.record_exception(e)
                 raise
```

Comment:
You can skip this `except` block from the coverage (that it's unnecessary to check) and reduce the number of error spans logged if you use the following:

```
    except Exception as exc:
        span.record_exception(exc)
else:
```

You can apply this for the rest of the code (and other files).

Reply:
Done.